google_guest_agent/non_windows_accounts.go

```diff
@@ -320,6 +323,43 @@ func readGoogleUsersFile() (map[string]string, error) {
 	return res, nil
 }
 
+// Enables AuthorizedKeysCommand to be used instead of an AuthorizedKeysFile
+func writeSSHDConfigFile() error {
+	sshConfig, err := os.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return err
+	}
+	proposed := filterAuthorizedKeyLines(string(sshConfig))
+	if proposed == string(sshConfig) {
+		return nil
+	}
+	file, err := os.OpenFile("/etc/ssh/sshd_config", os.O_WRONLY|os.O_TRUNC, 0777)
+	if err != nil {
+		return err
+	}
+	defer closeFile(file)
+	_, err = file.WriteString(proposed)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Helper function to set AuthorizedKeysCommand, should only run once in entire program execution
+func filterAuthorizedKeyLines(contents string) string {
```

Comment:
rather than cloning and editing the file editing functions, can we refactor the existing function so it can be used for both oslogin and metadata enablement modes?

Reply:
writeSSHDConfigFile() can probably be removed, and we can add an argument to the existing writeSSHConfigFile() function for this purpose.

I do not see any existing function which does what filterAuthorizedKeyLines() does, however. Is the suggestion to merge this functionality with something such as updateSSHConfig()? 

I'm thinking about calling something like filterAuthorizedKeyLines() instead of updateSSHConfig() in the metadata mode.